File: heranca/java/Aluno.java
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Herança em Java
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

public class Aluno extends Pessoa {
	private String curso;
	private String semestre;

	public Aluno() {
		System.out.println("Executando construtor da classe Aluno");
	}

	public String getCurso() {
		return curso;
	}

	public void setCurso(String curso) {
		this.curso = curso;
	}

	public String getSemestre() {
		return semestre;
	}

	public void setSemestre(String semestre) {
		this.semestre = semestre;
	}
}

File: heranca/cpp/Aluno.cpp
// Disciplina: Programação Orientada a Objetos
// Professor : Gilson Pereira
// Assunto   : Herança em C++
// Fonte     : Ascencio e Campos - Fundamentos da Programação de Computadores (3ª ed.)

#include "Aluno.hpp"
#include <iostream>

using namespace std;

Aluno::Aluno()
{
	cout << "\nExecutando construtor da classe Aluno";
}

string Aluno::getCurso()
{
    return curso;
}

void Aluno::setCurso(string c)
{
    curso = c;
}

string Aluno::getSemestre()
{
    return semestre;
}

void Aluno::setSemestre(string s)
{
    semestre = s;
}